src/main/scala/mesosphere/marathon/core/task/Task.scala

```diff
@@ -85,7 +85,7 @@ sealed trait Task {
   }
 
   def effectiveIpAddress(runSpec: RunSpec): Option[String] = {
-    if (runSpec.ipAddress.isDefined)
+    if (runSpec.container.exists(_.forceIp) || runSpec.ipAddress.isDefined)
```

Comment:
referring to `forceIp` isn't necessary here. For apps that use docker containers, it may be inferred that custom (USER) networking is in use by examining the network mode of the docker container.